spk pack: require nonempty appTitle and appMarketingVersion

dwrensha opened the pull request.

kentonv commented on src/sandstorm/spk.c++:
```diff
@@ -944,6 +948,15 @@ private:
     ensurePackageDefParsed();
 
+    if (!packageDef.getManifest().getAppTitle().hasDefaultText()) {
+      context.exitError(kj::str("Expected manifest.appTitle.defaultText to be nonempty."));
```
This error message feels like it may take more cognitive power than necessary to understand. Maybe a better one would be:

    Your package manifest is missing `appTitle`. In `sandstorm-pkgdef.capnp`, under `pkgdef.manifest`, add something like:
        appTitle = (defaultText = "My App"),

kentonv replied:
The `kj::str()` is redundant here.

dwrensha commented:
Updated. I moved the validation into `setPackageDef()`, where it can make use of the value of the `-p` flag if it was set. This additionally performs the validation for `spk dev`.

dwrensha commented:
(Note that the error message now automatically includes the name of the pkgdef file.)

kentonv commented:
Nice!
